extension/firefox/chrome/ShumwayCom.jsm

```diff
@@ -273,6 +291,98 @@ var ShumwayCom = {
           return;
         }
         onSystemResourceCallback = callback;
+      },
+      createLocalConnection: function(connectionName, callback) {
+        if (typeof connectionName !== 'string' ||
+            connectionName[0] !== '_' && connectionName.split(':').length !== 2) {
+          return -1; // LocalConnectionConnectResult.InvalidName
+        }
+        if (localConnectionsRegistry[connectionName]) {
+          return -2; // LocalConnectionConnectResult.AlreadyTaken
+        }
+        var connection = {
+          callback,
+          allowedSecureDomains: Object.create(null),
+          allowedInsecureDomains: Object.create(null)
+        };
+        localConnectionsRegistry[connectionName] = connection;
+        return 0; // LocalConnectionConnectResult.Success
+      },
+      hasLocalConnection: function(connectionName) {
+        return !!_getLocalConnection(connectionName);
+      },
+      closeLocalConnection: function(connectionName) {
+        if (!_getLocalConnection(connectionName)) {
+          return -1; // LocalConnectionCloseResult.NotConnected
+        }
+        delete localConnectionsRegistry[connectionName];
+        return 0; // LocalConnectionCloseResult.Success
+      },
+      sendLocalConnectionMessage: function(connectionName, methodName, argsBuffer, sender,
+                                           senderURL) {
```

Comment:
Right, here's my idea for that: when we load a SWF into a player instance, we note the URL In `sendLocalConnectionMessage`, we check if `senderURL` is in the set of remembered URLs for that player instance. If not, we reject here. Do you think that makes sense?